tests/test_cases/fchdir.c

```diff
@@ -0,0 +1,58 @@
+#define _GNU_SOURCE
+
+#include <fcntl.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+
+#define MAX_PATH 1024
+
+int main(void) {
+    char path[MAX_PATH];
+    
+    // Get the current working directory
+    char* result = getcwd(path, sizeof(path));
+    if (result == NULL) {
+        perror("getcwd() error");
+        return EXIT_FAILURE;
+    } else {
+        printf("current working directory is: %s :: %s\n", path, result);
+    	fflush(stdout);
+    }
+
+    // Open a directory
+    int fd = open("automated_tests/", O_RDWR);
+    if (fd == -1) {
+        perror("Error with open");
+        return EXIT_FAILURE;
+    }
+
+    // Change the current directory to the directory referred to by the open file descriptor
+    if (fchdir(fd) == -1) {
+        perror("Error with fchdir");
+        close(fd);
+        return EXIT_FAILURE;
+    }
+
+    // Get the current working directory
+    char* second_result = getcwd(path, sizeof(path));
+    if (second_result == NULL) {
+        perror("Error with getcwd");
+        close(fd);
+        return EXIT_FAILURE;
+    } else {
```

Comment:
Since the `if` path returns a value, maybe we can get rid of the `else` clause? This would improve clarity while reading the code.